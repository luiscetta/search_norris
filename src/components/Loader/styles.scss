.loader {
    position: fixed;
    top: 75%;
    left: 50%;
    transform: translate(-50%, -50%);
}