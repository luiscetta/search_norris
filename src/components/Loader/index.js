import React from 'react';
import { ProgressBar as Loader } from 'react-loader-spinner';

import './styles.scss';


export default function Loading() {
    return (
        <div className="loader">
            <Loader color="#fff"/>
        </div>
    )
}
