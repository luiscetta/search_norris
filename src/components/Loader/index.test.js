import React from 'react';

import { create } from 'react-test-renderer';

import Loader from './';


describe('Loader component', () => {
    describe('When the component is rendered', () => {
        it('Should match snapshot', () => {
            const wrapper = create(<Loader />);
            expect(wrapper.toJSON()).toMatchSnapshot();
        });
    })
});