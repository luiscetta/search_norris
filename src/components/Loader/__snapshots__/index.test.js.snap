// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Loader component When the component is rendered Should match snapshot 1`] = `
<div
  className="loader"
>
  <svg
    aria-busy={true}
    aria-label="progress-bar-loading"
    className=""
    data-testid="progress-bar-svg"
    height="80"
    preserveAspectRatio="xMidYMid"
    role="status"
    style={Object {}}
    viewBox="0 0 100 100"
    width="80"
    xmlns="http://www.w3.org/2000/svg"
  >
    <defs>
      <clipPath
        height="100"
        id="lds-progress-cpid-5009611b8a418"
        ng-attr-id="{{config.cpid}}"
        width="100"
        x="0"
        y="0"
      >
        <rect
          height="100"
          width="66.6667"
          x="0"
          y="0"
        >
          <animate
            attributeName="width"
            begin="0s"
            calcMode="linear"
            dur="1"
            keyTimes="0;0.5;1"
            repeatCount="indefinite"
            values="0;100;100"
          />
          <animate
            attributeName="x"
            begin="0s"
            calcMode="linear"
            dur="1"
            keyTimes="0;0.5;1"
            repeatCount="indefinite"
            values="0;0;100"
          />
        </rect>
      </clipPath>
    </defs>
    <path
      d="M82,63H18c-7.2,0-13-5.8-13-13v0c0-7.2,5.8-13,13-13h64c7.2,0,13,5.8,13,13v0C95,57.2,89.2,63,82,63z"
      fill="none"
      ng-attr-stroke="{{config.c1}}"
      stroke="#F4442E"
      strokeWidth="2.7928"
    />
    <path
      clipPath="url(#lds-progress-cpid-5009611b8a418)"
      d="M81.3,58.7H18.7c-4.8,0-8.7-3.9-8.7-8.7v0c0-4.8,3.9-8.7,8.7-8.7h62.7c4.8,0,8.7,3.9,8.7,8.7v0C90,54.8,86.1,58.7,81.3,58.7z"
      fill="#51E5FF"
      ng-attr-clip-path="url(#{{config.cpid}})"
      ng-attr-fill="{{config.c2}}"
    />
  </svg>
</div>
`;
